lib/console.c

```diff
@@ -180,10 +205,16 @@ console_print_box_at(CHAR16 *str_arr[], int highlight,
 		return;
 	}
 
-	SetMem16 (Line, size_cols * 2, BOXDRAW_HORIZONTAL);
+	/* test if boxdraw characters work */
+	co->SetCursorPosition(co, start_col, start_row);
+	Line[0] = boxdraw[0].up_left;
+	Line[1] = L'\0';
+	char_set = co->OutputString(co, Line) == 0 ? 0 : 1;
```

Comment:
UEFI Specification Version 2.8 Errata B does specify the BOXDRAW characters as mandatory. As per the Spec's page 448, the Status Codes that can be returned here are EFI_SUCCESS, EFI_DEVICE_ERROR, EFI_UNSUPPORTED, and EFI_WARN_UNKNOWN_GLYPH.

I would prefer to test for EFI_WARN_UNKNONW_GLYPH (and if that's in fact what is returned on ASRock) and then switch to the other character set based on that.

Also it would be nice to retest that the one one works and doesn't also return EFI_WARN_UNKNOWN_GLYPH in case the firware just returns that always irrespective if it was successful or not.